### PR TITLE
BaSP-TG: for projects

### PR DESCRIPTION
**Pull request by sabrina-reckinger**
the JSON data was created and the modify,  delete and filter by part of the name functions was created.

### CODE REVIEW

File: src/resources/projects.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if (err) {
+            res.send("Cannot delete project")
+        } else {
+            res.send("Project deleted")
+        }
+    })
+})
+
+router.post("/modify/:id", (req, res) => {
+    const changeProject = req.body
```
Comment (ignacioaristo) [inline src/resources/projects.js:29]: ; is missing

Comment (sabrina-reckinger) [reply src/resources/projects.js]: fixed

File: src/resources/projects.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const express = require("express");
+const router = express.Router();
+const projects = require("../data/projects.json");
+const fs = require('fs')
```
[the remark this answers is not in the document]
Comment (sabrina-reckinger) [reply src/resources/projects.js]: fixed

**Comment by sabrina-reckinger** [comment]
fixed all ;

File: src/index.js [hunk 3]
```diff
@@ -4,10 +4,14 @@ import express from 'express';
 // use "require" to import JSON files
 const admins = require('./data/admins.json');
 
+
```
Comment (ignacioaristo) [inline src/index.js:7]: Delete this empty line

File: src/index.js [hunk 4]
```diff
@@ -4,10 +4,14 @@ import express from 'express';
...
+
```
Comment (ignacioaristo) [inline src/index.js:8]: Delete this empty line

File: src/resources/projects.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        if (err) {
+            res.send("Cannot delete project");
+        } else {
+            res.send("Project deleted");
+        }
+    });
+});
+router.post("/modify/:id", (req, res) => {
+    const changeProject = req.body;
+    const projectId = parseInt(req.params.id);
```
Comment (ignacioaristo) [inline src/resources/projects.js:28]: We need to add a second parameter in `parseInt` method with value 10.

File: src/resources/projects.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        if (err) {
+            res.send("Cannot modify project");
+        } else {
+            res.send("Project modified");
+        }
+    });
+});
+router.post("/addEmployee/:id", (req, res) => {
+    const newEmployee = req.body;
+    const projectId = parseInt(req.params.id);
```
Comment (ignacioaristo) [inline src/resources/projects.js:41]: The same here.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (sabrina-reckinger) [reply src/index.js:7]: done

File: src/index.js (hunk 4 above)
Comment (sabrina-reckinger) [reply src/index.js:8]: done

File: src/resources/projects.js (hunk 5 above)
Comment (sabrina-reckinger) [reply src/resources/projects.js:28]: done

File: src/resources/projects.js (hunk 6 above)
Comment (sabrina-reckinger) [reply src/resources/projects.js:41]: done

File: src/index.js [hunk 7]
```diff
@@ -7,22 +7,22 @@ const tasksRouter = require('./resources/tasks');
...
 const superAdmin = require('./resources/super-admins');
-
 const app = express();
+const projectRouter = require("./resources/projects");
 const port = process.env.PORT || 3000;
+const tasksRouter = require("./resources/tasks");
 
 app.use(express.json());
+app.use("/projects", projectRouter);
+app.use("/tasks", tasksRouter);
```
Comment (paurinaldi) [inline src/index.js:17]: delete duplicated line

File: src/index.js [hunk 8]
```diff
@@ -7,22 +7,22 @@ const tasksRouter = require('./resources/tasks');
...
 app.use('/tasks', tasksRouter);
 app.use('/timeSheets', timeSheetsRouter);
 app.use('/employees', employees);
 app.use('/superAdmin', superAdmin);
-
 app.use(projectRouter);
```
Comment (paurinaldi) [inline src/index.js:22]: delete this line, it's incomplete and duplicated

File: src/resources/projects.js [hunk 9]
```diff
@@ -1,9 +1,61 @@
...
+    fs.writeFile('src/data/projects.json', JSON.stringify(newProject), (err) => {
+        if (err) {
+            res.send("Cannot add employee to the project");
+        } else {
+            res.send("Employee added to the project");
+        }
+    });
+});
+module.exports = router;
 const fs = require('fs');
```
Comment (paurinaldi) [inline src/resources/projects.js:55]: delete duplicated lines

**Review by paurinaldi — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 7 above)
Comment (sabrina-reckinger) [reply src/index.js:17]: done

File: src/resources/projects.js (hunk 9 above)
Comment (sabrina-reckinger) [reply src/resources/projects.js:55]: done

File: src/index.js (hunk 8 above)
Comment (sabrina-reckinger) [reply src/index.js:22]: done

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
We have a lot of problems in the package-lock.json.
Delete this file, run npm install and update your local master with the origin.
If you have conflicts, keep the incoming changes.

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**